oci/spec_opts.go

```diff
@@ -683,6 +683,10 @@ func WithUser(userstr string) SpecOpts {
 				return err
 			}
 
+			// Use a read-only mount when trying to get user/group information
+			// from the container's rootfs. Since the option does read operation
+			// only, we append ReadOnly mount option to prevent the Linux kernel
+			// from syncing whole filesystem in umount syscall.
```

Comment:
@fuweid let me know if this is enough to restore, or if the second half (the TODO) of the original comment is still relevant in some cases;

```go
// TODO(fuweid):
//
// Currently, it only works for overlayfs. I think we can apply it to other
// kinds of filesystem. Maybe we can return `ro` option by `snapshotter.Mount`
// API, when the caller passes that experimental annotation
// `containerd.io/snapshot/readonly.mount` something like that.
```

Reply:
Thanks! It looks good to me.

Reply:
Thanks! Wanted to be sure if there wasn't some case I overlooked ☺️❤️